py/sentry_streams/flink/flink_adapter.py

```diff
@@ -29,6 +29,30 @@ def __init__(self, config: MutableMapping[str, Any], env: StreamExecutionEnviron
         self.environment_config = config
         self.env = env
 
+    def load_function(self, step: Step) -> Any:
+        """
+        Takes a transform step containing a function, and either returns
+        the function (if it's a Callable) or dynamically loads and returns the
+        function (if it's a path to a module).
+        """
+        # TODO: break out the dynamic loading logic into a
+        # normalization layer before the flink adapter
+        assert hasattr(step, "function"), "load_function() requires a step containing a function."
+        if isinstance(step.function, str):
```

Comment:
Please let's not turn this back to dynamic loading.
`hasattr(step, "function")` 

There are ways to solve this with proper typing:

1. `step: Union[Map, Filter]`. This makes `step.function` statically typed.
2. Create a class on top of Fitler and Map they inherit the `function` attribute from. Then `load_function` can take an object of that type.